ddtrace/tracer/payload.go

```diff
@@ -108,9 +113,20 @@ func (p *payload) updateHeader() {
 func (p *payload) Read(b []byte) (n int, err error) {
 	if p.off < len(p.header) {
 		// reading header
-		n = copy(b, p.header[p.off:])
+		n, err := p.buf.Write(p.header[p.off:])
+		if err != nil {
+			return n, err
+		}
 		p.off += n
-		return n, nil
+	}
+	for len(p.traces) != 0 && p.buf.Len() <= len(b) {
+		// fill buffer
+		msgp.Encode(&p.buf, p.traces[0])
```

Comment:
Looks like we forgot to set the error:

    err = msgp.Encode(&p.buf, p.traces[0])